elsarticle-template-5-harv.tex

```diff
@@ -391,33 +389,33 @@ \section{Platform Design}
-        This engages the mechanical park brakes, removes all tractive effort from the motor controllers, and de-powers mounted modules.
+        This engages the mechanical park brakes, removes all tractive effort from the motors, and de-powers mounted modules.
 
-        The open source Robotic Operating System (ROS) is used to facilitate communication between computers and software nodes running within each computer.
+        The open source Robotic Operating System (ROS) is used to facilitate communication between computers and software nodes running within each computer (\cite{quigley2009}).
         Nodes written using this framework follow either a publish-subscribe or service-client pattern.
         To maximise code reusability, each device on the platform has its own ROS-compatible node dedicated to publishing device data or subscribing to generated device commands.
-        Interface adapters, motor controllers, wireless controllers, lidar, and encoders are examples of devices with dedicated interface nodes.
-        Nodes are also used to transform or perform calculations on data while passing it between nodes written in either C++ or Python.
-        For instance, as shown in Figure~\ref{fig:system_diagram_software}, an `Ackermann kinematics' node transforms steering input data into individual wheel velocity and position/angle outputs.
-        Among other things, ROS offers the ability to monitor and record all communication passing through it which can be replayed and interrogated at a later date.
+        Interface adapters, motor controllers, wireless controllers, lidar, and encoders are examples of devices on the platform with dedicated interface nodes.
+        Nodes can be written in either C++ or Python and can be used simply to transform or perform calculations on data while passing it between other nodes.
+        For instance, as shown in Figure~\ref{fig:system_diagram_software}, an `Ackermann kinematics' node transforms a steering vector into individual wheel velocity and position/angle outputs.
+        Among other things, ROS offers the ability to monitor and record all communication passing through it which can be replayed and examined later.
 
         The manufacturer's configuration of the motor controllers require them to be interfaced using a combination of analogue and digital inputs.
         For example, the accelerator and steering inputs are controlled by potentiometers actuated by the vehicle's driver.
         However, the controllers also provide an option for a multi-motor vehicle configuration.
-        In this configuration, the analogue inputs fed into a master controller are relayed to a second controller over a CAN interface.
+        In this configuration, the analogue inputs fed into a master controller are relayed to a slave, controller over a CAN interface.
```

Comment:
Unnecessary comma. 
```suggestion
        In this configuration, the analogue inputs fed into a master controller are relayed to a slave controller over a CAN interface.
```
Also, my thinking is that by saying "slave controller" only is that the reader may interpret this to mean that it is a different kind of controller. It's not bullet-proof because I've used the term master but for me at least it reduces ambiguity a little. Slave controller could also mean another master that controls slave devices.
```suggestion
        In this configuration, the analogue inputs fed into a master controller are relayed to a second (slave) controller over a CAN interface.
```

Reply:
Good point. I like the change.